ansible/shell_plugins/docker.py

```diff
@@ -29,7 +41,22 @@ def join_path(self, *args):
 
         return super(ShellModule, self).join_path(*args)
 
-    def build_module_command(self, env_string, shebang, cmd, arg_path=None, rm_tmp=None):
+    def remove(self, path, recurse=False):
+        if not self.container:
+            self.container = self.get_option('ansible_python_interpreter').split()[3]
```

Comment:
>self.get_option('ansible_python_interpreter').split()[3] [](start = 29, length = 56)

This logic is not robust.
If this case `not self.container` happens in your test? If not, we can just fallback to base class function.

Reply:
If I am right this module was made just for one thing, to execute tasks inside the container on the host. And it retrieves the container name from shebang in `build_module_command` which is not anything else as  `ansible_python_interpreter`. And for this plugin, it should look like `docker exec -i  <container> python`  what is not obvious. So I just put `get_option('ansible_python_interpreter')` under `if` to minimize `get_option` calls.

For any case changed this to the same approach as in `build_module_command`.

For me, it looks like is better to define mandatory `ansible_container` option for this plugin, change `remove` and `build_module_command` to use it as a container name source and replace 
```
vars:
    ansible_shell_type: docker
    ansible_python_interpreter: docker exec -i swss python
```
to 
```
vars:
    ansible_shell_type: docker
    ansible_container: swss
```
I could do this in separate PR if you accept.